test/integration/test-commands.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+import puppeteer from "puppeteer";
+import test from "node:test";
+import assert from "node:assert"
+import {initializeExtension, sleep} from "../utils.js";
+
+test("Extension command to functionality test", async (t) => {
+    var browser = await puppeteer.launch(); await browser.close();
+    var service_worker;
+    t.beforeEach(async (t) => {
+        [browser, service_worker] = await initializeExtension();
+    });
+    t.afterEach(async (t) => {
+        await browser.close();
+    });
+
+    await t.test("Test content-script Single Switching", async (t) => {
+        const page1 = await browser.newPage();
+        await page1.bringToFront();
+        await page2.goto("chrome://extensions/", {waitUntil: 'domcontentloaded'});
+        const page2 = await browser.newPage();
+        await page2.bringToFront();
+        await page2.goto("chrome://extensions/", {waitUntil: 'domcontentloaded'});
+
+        assert.notStrictEqual(await getActivePage(browser, 3000), page1, "expected us to be on page 2");
+        assert.strictEqual(await getActivePage(browser, 3000), page2, "expected us to be on page 2");
+
+        await page2.keyboard.down("Control");
+
+        await sleep(100);
+
+        await page2.keyboard.down("KeyQ");
+
+        await sleep(100);
+
+        await page2.keyboard.up('Control');
+
+        await sleep(100);
+
+        await page2.keyboard.up("KeyQ");
+
+        await sleep(100);
+
+        assert.strictEqual(await getActivePage(browser, 3000), page1, "we don't end on page 1");
+    });
+    
+    await t.test("Test content-script single Multiple Switching", async (t) => {
+        const page1 = await browser.newPage();
+        await page1.bringToFront();
+        const page2 = await browser.newPage();
+        await page2.goto("https://example.com", {waitUntil: 'domcontentloaded'});
+        await page2.bringToFront();
+
+        assert.notStrictEqual(await getActivePage(browser, 3000), page1, "expected us to be on page 2");
+        assert.strictEqual(await getActivePage(browser, 3000), page2, "expected us to be on page 2");
```

Comment:
Nit: Maybe consider `assert.notDeepStrictEqual` and `assert.deepStrictEqual` as they would dive into the individual parts of the page. Probably not necessary as the pages would probably have some ID but just in case they have the same looking fields.